docs/articles_en/learn-openvino/llm_inference_guide/genai-guide.rst

```diff
@@ -0,0 +1,277 @@
+
+         int main(int argc, char* argv[]) {
+            std::string model_path = argv[1];
+            ov::LLMPipeline pipe(model_path, "CPU");
+            ov::GenerationConfig config = pipe.get_generation_config();
+            config.max_new_tokens = 256;
+            config.num_groups = 3;
+            config.group_size = 5;
+            config.diversity_penalty = 1.0f;
+
+            cout << pipe.generate("The Sun is yellow because", config);
+         }
+
+Specify generation_config to use grouped beam search:
+
+.. tab-set::
+
+   .. tab-item:: C++
+
+      .. code-block:: cpp
+
+         int main(int argc, char* argv[]) {
+            std::string prompt;
+
+            std::string model_path = argv[1];
+            ov::LLMPipeline pipe(model_path, "CPU");
+
+            ov::GenerationConfig config = pipe.get_generation_config();
+            config.max_new_tokens = 256;
+            config.num_groups = 3;
+            config.group_size = 5;
+            config.diversity_penalty = 1.0f;
+
+            auto streamer = [](std::string word) { std::cout << word << std::flush; };
+
+            pipe.start_chat();
+            for (size_t i = 0; i < questions.size(); i++) {
+
+               std::cout << "question:\n";
+               cout << prompt << endl;
+
+               auto answer = pipe(prompt, config, streamer);
+               // no need to print answer, streamer will do that
+            }
+            pipe.finish_chat();
+         }
+
+Comparing with Hugging Face Results
+#######################################
+
+Compare and analyze results with those generated by Hugging Face models.
+
+.. tab-set::
+
+   .. tab-item:: Python
+
+      .. code-block:: python
+
+         from transformers import AutoTokenizer, AutoModelForCausalLM
+
+         tokenizer = AutoTokenizer.from_pretrained("TinyLlama/TinyLlama-1.1B-Chat-v1.0")
+         model = AutoModelForCausalLM.from_pretrained("TinyLlama/TinyLlama-1.1B-Chat-v1.0")
+
+         max_new_tokens = 32
+         prompt = 'table is made of'
+
+         encoded_prompt = tokenizer.encode(prompt, return_tensors='pt', add_special_tokens=False)
+         hf_encoded_output = model.generate(encoded_prompt, max_new_tokens=max_new_tokens, do_sample=False)
+         hf_output = tokenizer.decode(hf_encoded_output[0, encoded_prompt.shape[1]:])
+         print(f'hf_output: {hf_output}')
+
+         import sys
+         sys.path.append('build-Debug/')
+         import py_generate_pipeline as genai # set more friendly module name
+
+         pipe = genai.LLMPipeline('text_generation/causal_lm/TinyLlama-1.1B-Chat-v1.0/pytorch/dldt/FP16/')
+         ov_output = pipe(prompt, max_new_tokens=max_new_tokens)
+         print(f'ov_output: {ov_output}')
+
+         assert hf_output == ov_output
+
+GenAI API
+#######################################
+
+OpenVINO GenAI Flavor includes the following API methods:
+
+* generation_config
+* llm_pipeline
+* streamer_base
+* tokenizer
+* visibility
```

Comment:
These are headers, not methods. Is it possible to build docs based on docstrings similar to https://docs.openvino.ai/2022.3/classov_1_1Core.html#detailed-documentation

Reply:
yes but let's plan to implement this after the release, so we can merge and publish the initial PR with install and descriptions 
@kblaszczak-intel to confirm

Reply:
Sure! Let's merge and extend later. We'll need time to do it manually, as for now we're unable to generate API documents from source for genAI.

Reply:
API is expected to be heavily restructured by the next release. I'd like to warn you that you will end up doing it again.